### PR TITLE
feat: no buffer req. rewrite

### PR DESCRIPTION
**Pull request by TateB**

### CODE REVIEW

**Comment by ukstv** [comment]
@TateB 2 cents from an almost bystander. The code becomes much-much easier to leverage https://github.com/paulmillr/scure-base and `Coder` interface there. 80% of the plumbing goes away with that.

**Comment by TateB** [comment]
> @TateB 2 cents from an almost bystander. The code becomes much-much easier to leverage https://github.com/paulmillr/scure-base and `Coder` interface there. 80% of the plumbing goes away with that.

completely forgot about that, thanks! will use it

**Comment by ukstv** [comment]
@TateB Is this library used for high-performance processing? IMO, its main use case is on UI where speed is not that important. Size is quite important though.

Anyhow, that's what pissed _me_ off about the current state of the library. I was disappointed enough to do a rewrite myself. Considering how PRs are merged here, it's a separate repo: https://github.com/ukstv/ens-address-encoder maybe you'll find it useful.

**Comment by ukstv** [comment]
Rooting for this to land in prod, anyway.

**Comment by TateB** [comment]
> Is this library used for high-performance processing? IMO, its main use case is on UI where speed is not that important. Size is quite important though.

not internally, but in the future it's definitely possible. performance isn't really a focus as much as it is a nice to have, but using the `@scure/base` lib (when most of the funcs being used from it were already written) led to a 5-10x performance decrease.

**Review by LeonmanRolls — APPROVED** [review]
Great work!

**Comment by talentlessguy** [comment]
Would it be possible to publish this under `alpha` tag?

**Comment by talentlessguy** [comment]
I can't decode a BTC address:

```js
import { createPublicClient, http } from 'viem'

const publicClient = createPublicClient({
  transport: http('https://rpc.ankr.com/eth'),
  chain: mainnet,
})

import { mainnet } from 'viem/chains'
import { getCoderByCoinType } from '@ensdomains/address-encoder'

for (const coinType of resolver.coinTypes) {
  const hexAddress = await publicClient.getEnsAddress({
    coinType: parseInt(coinType),
    name,
  })

  const coin = getCoderByCoinType(parseInt(coinType))

  if (hexAddress) {
    data.coins.push({
      type: coin.name,
      address:
        coin.coinType === 60
          ? hexAddress
          : td.decode(coin.decode(hexAddress.slice(2))),
    })
  }
}
```

it parses ethereum fine but on btc this happens:

```
⨯ Error: Invalid checksum
```

the hex address (returned by viem) is `0x001499e3452752d781c06aa4fb828e85ea7d3b57ff1b`. I tried slicing 0x, and keeping it, same result. Am I doing something wrong?

File: README.md [hunk 1]
```diff
@@ -12,179 +12,84 @@ Please read [the specification page for the more detail](https://docs.ens.domain
...
+const btcCoder = getCoderByCoinName("btc");
 
+// Convert hex encoded data to bytes
+const dataAsBytes = hexToBytes(
+  "0x76a91462e907b15cbf27d5425399ebf6f0fb50ebb88f1888ac"
+);
+// Uint8Array(25) [ 118, 169, 20, 98, 233, 7, 177, 92, 191, 39, 213, 66, 83, 153, 235, 246, 240, 251, 80, 235, 184, 143, 24, 136, 172 ]
+
+// Pass bytes to encoder
+const encodedAddress = btcCoder.encode(decodedAddress);
```
Comment (sugh01) [inline README.md:54]: `const encodedAddress = btcCoder.encode(dataAsBytes);`

**Pull request merged**